aspnetcore/security/authentication/cookie.md

```diff
@@ -14,335 +14,369 @@ uid: security/authentication/cookie
 
-## Adding and configuring
+## Configuration
 
 # [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-Complete the following steps:
+If you aren't using the [Microsoft.AspNetCore.All metapackage](xref:fundamentals/metapackage), install version 2.0+ of the [Microsoft.AspNetCore.Authentication.Cookies](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Cookies/) NuGet package.
 
-- If not using the `Microsoft.AspNetCore.All` [metapackage](xref:fundamentals/metapackage), install version 2.0+ of the `Microsoft.AspNetCore.Authentication.Cookies` NuGet package in your project.
+In the `ConfigureServices` method, create the Authentication Middleware service with the `AddAuthentication` and `AddCookie` methods:
 
-- Invoke the `UseAuthentication` method in the `Configure` method of the *Startup.cs* file:
+[!code-csharp[Main](cookie/sample/Startup.cs?name=snippet1)]
 
-    ```csharp
-    app.UseAuthentication();
-    ```
+`AuthenticationScheme` passed to `AddAuthentication` sets the authentication scheme. `AuthenticationScheme` is useful when there are multiple instances of cookie authentication and you want to [limit authorization to one instance](xref:security/authorization/limitingidentitybyscheme). Setting the `AuthenticationScheme` to `CookieAuthenticationDefaults.AuthenticationScheme` provides a value of "Cookies" for the scheme. You can supply any string value that distinguishes the scheme.
 
-- Invoke the `AddAuthentication` and `AddCookie` methods in the `ConfigureServices` method of the *Startup.cs* file:
+In the `Configure` method, use the `UseAuthentication` method to invoke the Authentication Middleware that sets the `HttpContext.User` property. Call the `UseAuthentication` method before calling `AddMvcWithDefaultRoute` in an MVC app or `AddMvc` in a Razor Pages app:
 
-    ```csharp
-    services.AddAuthentication("MyCookieAuthenticationScheme")
-            .AddCookie("MyCookieAuthenticationScheme", options => {
-                options.AccessDeniedPath = "/Account/Forbidden/";
-                options.LoginPath = "/Account/Unauthorized/";
-            });
-    ```
+[!code-csharp[Main](cookie/sample/Startup.cs?name=snippet2)]
 
-# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+**AddCookie Options**
 
-Complete the following steps:
+The [CookieAuthenticationOptions](/dotnet/api/microsoft.aspnetcore.authentication.cookies.cookieauthenticationoptions?view=aspnetcore-2.0) class is used to configure the authentication provider options.
 
-- Install the `Microsoft.AspNetCore.Authentication.Cookies` NuGet package in your project. This package contains the cookie middleware.
+| Option | Description |
+| ------ | ----------- |
+| [AccessDeniedPath](/dotnet/api/microsoft.aspnetcore.authentication.cookies.cookieauthenticationoptions.accessdeniedpath?view=aspnetcore-2.0) | Informs the handler that it should change an outgoing 403 Forbidden status code into a 302 Found (URL redirect) with the provided path. The default value is `/Account/AccessDenied`. |
```

Comment:
This does not react to 403s, it is only triggered by HttpContext.ForbidAsync().

Reply:
hum ... this isn't good then :point_right: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.cookies.cookieauthenticationoptions.accessdeniedpath?view=aspnetcore-2.0